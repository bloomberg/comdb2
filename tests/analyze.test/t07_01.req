select 1
create table t7 {
schema 
{
    int a
    int b
    int c
    int d
}

keys 
{
    "key" = a+b
}
}
$$
insert into t7 values(1,1,1,1)
insert into t7 values(2,1,1,1)
insert into t7 values(3,1,1,1)
insert into t7 values(4,1,1,1)

analyze t7
select tbl,substr(idx,1,length(idx)-8) as idx,stat from sqlite_stat1 where tbl='t7' order by idx

alter table t7 {
schema 
{
    int a
    int b
    int c
    int d
}

keys 
{
    "key_ab" = a+b
}
}
$$
select tbl,substr(idx,1,length(idx)-8) as idx,stat from sqlite_stat1 where tbl='t7' order by idx
select 1
