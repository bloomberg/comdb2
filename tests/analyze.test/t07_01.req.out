(1=1)
[select 1] rc 0
[create table t7 {
schema 
{
    int a
    int b
    int c
    int d
}

keys 
{
    "key" = a+b
}
}] rc 0
(rows inserted=1)
[insert into t7 values(1,1,1,1)] rc 0
(rows inserted=1)
[insert into t7 values(2,1,1,1)] rc 0
(rows inserted=1)
[insert into t7 values(3,1,1,1)] rc 0
(rows inserted=1)
[insert into t7 values(4,1,1,1)] rc 0
[analyze t7] rc 0
(tbl='t7', idx='$KEY_', stat='4 1 1')
[select tbl,substr(idx,1,length(idx)-8) as idx,stat from sqlite_stat1 where tbl='t7' order by idx] rc 0
[alter table t7 {
schema 
{
    int a
    int b
    int c
    int d
}

keys 
{
    "key_ab" = a+b
}
}] rc 0
(tbl='t7', idx='$KEY_', stat='4 1 1')
(tbl='t7', idx='$KEY_AB_', stat='4 1 1')
[select tbl,substr(idx,1,length(idx)-8) as idx,stat from sqlite_stat1 where tbl='t7' order by idx] rc 0
(1=1)
[select 1] rc 0
